src/NuGet.Core/NuGet.Protocol.Core.v3/Resources/PackageMetadataResourceV3.cs

```diff
@@ -29,8 +28,13 @@ public PackageMetadataResourceV3(HttpSource client, RegistrationResourceV3 regRe
 
         public override async Task<IEnumerable<IPackageSearchMetadata>> GetMetadataAsync(string packageId, bool includePrerelease, bool includeUnlisted, Common.ILogger log, CancellationToken token)
         {
-            var metadataList = await _regResource.GetPackageMetadata(packageId, includePrerelease, includeUnlisted, log, token);
-            return metadataList.Select(ParseMetadata);
+            var metadataCache = new MetadataReferenceCache();
+
+            return
```

Comment:
assign this to a var before returning it like before. It makes it easier to debug.